src/fsharp/FSharp.Core/array.fs

```diff
@@ -1136,23 +1128,22 @@ namespace Microsoft.FSharp.Collections
             [<CompiledName("Partition")>]
             let partition predicate (array : 'T[]) =
                 checkNonNull "array" array
-                let inputLength = array.Length
-                let lastInputIndex = inputLength - 1
+                let inputLength = array.Length                
 
                 let isTrue = Array.zeroCreateUnchecked inputLength
                 Parallel.For(0, inputLength, 
                     fun i -> isTrue.[i] <- predicate array.[i]
                     ) |> ignore
                 
                 let mutable trueLength = 0
-                for i in 0 .. lastInputIndex do
+                for i in 0 .. isTrue.Length-1 do
```

Comment:
I couldn't say. `0 .. bla` it's a range expression so you can write fancier things, `0 to bla` is the low-level for loop that doesn't rely on too much compiler optimizations to be good. I prefer writing the latter when I want to make it clear I'm writing for performance.